code/Forms/EditorExternalLinkFormFactory.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace SilverStripe\Admin\Forms;
+
+use InvalidArgumentException;
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Forms\CheckboxField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormAction;
+use SilverStripe\Forms\FormFactory;
+use SilverStripe\Forms\TextField;
+
+class EditorExternalLinkFormFactory implements FormFactory
+{
+    use Extensible;
+    use Injectable;
+    use Configurable;
+    
+    /**
+     * @param RequestHandler $controller
+     * @param string $name
+     * @param array $context
+     * @return Form
+     */
+    public function getForm(RequestHandler $controller = null, $name = FormFactory::DEFAULT_NAME, $context = [])
+    {
+        // Validate context
+        foreach ($this->getRequiredContext() as $required) {
+            if (!isset($context[$required])) {
+                throw new InvalidArgumentException("Missing required context $required");
+            }
+        }
+    
+        $fields = $this->getFormFields($controller, $name, $context);
+        $actions = $this->getFormActions($controller, $name, $context);
+        $validator = $this->getValidator($controller, $name, $context);
+        /** @var Form $form */
+        $form = Form::create($controller, $name, $fields, $actions, $validator);
+        $form->addExtraClass('form--no-dividers');
+        
+        return $form;
+    }
+    
+    protected function getFormFields($controller, $name, $context)
+    {
+        $fields = FieldList::create([
+            TextField::create('Link', _t('SilverStripe\\Forms\\HTMLEditor\\HTMLEditorField.URL', 'URL'), 'http://'),
+            TextField::create(
+                'Description',
+                _t('SilverStripe\\Forms\\HTMLEditor\\HTMLEditorField.LINKDESCR', 'Link description')
```

Comment:
I think this could lead to many duplication instances, but maybe I'm just paranoid.
I wouldn't be completely opposed to it.
e.g. I prefer `_t(LeftAndMain::class.'.something')` over `_t(self::class.'.something')` or `_t(__CLASS__.'.something')` for clarity.